### PR TITLE
BaSP-MR-48: Super Admin TEST

### PR DESCRIPTION
**Pull request by ukiweskamp**

### CODE REVIEW

File: src/tests/super-admin.test.js [hunk 1]
```diff
@@ -1,3 +1,155 @@
...
+import request from 'supertest';
+import app from '../app';
+import SuperAdmin from '../models/Super-admin';
+import superadminSeed from '../seeds/super-admin';
+
+beforeAll(async () => {
+  await SuperAdmin.collection.insertMany(superadminSeed);
+});
+
+describe('GET /api/Super-admin', () => {
```
Comment (DelfiLT) [inline src/tests/super-admin.test.js:10]: Maybe you could test if the data is empty or not

File: src/tests/super-admin.test.js [hunk 2]
```diff
@@ -1,3 +1,155 @@
...
+
+  test('POST Invalid post data should return status 400', async () => {
+    const mocksuperAdmin = {
+      email: 'supmin.com',
+      password: 'rip123!',
+    };
+    const response = await request(app).post('/api/super-admin').send(mocksuperAdmin);
+    expect(response.status).toBe(400);
+    expect(response.body.message).toBeDefined();
+    // eslint-disable-next-line no-underscore-dangle
```
Comment (DelfiLT) [inline src/tests/super-admin.test.js:72]: try not to disable eslint

File: src/tests/super-admin.test.js [hunk 3]
```diff
@@ -1,3 +1,155 @@
...
+    expect(response.error).toBeTruthy();
```
Comment (DelfiLT) [inline src/tests/super-admin.test.js:73]: Be careful with the response.error, should be response.body.error

Comment (DelfiLT) [reply src/tests/super-admin.test.js:73]: Check this in every case

File: src/tests/super-admin.test.js [hunk 4]
```diff
@@ -1,3 +1,155 @@
...
+    const reqBody = {
+      email: 'maradonaa@gmail.com',
+    };
+    const response = await request(app).put('/api/super-admin/64619de9d65e6b69280df4c9').send(reqBody);
+    expect(response.status).toBe(400);
+    expect(response.body.message).toBeDefined();
+    expect(response.body.error).toBeTruthy();
+  });
+
+  test('PUT if update success return 201', async () => {
```
Comment (DelfiLT) [inline src/tests/super-admin.test.js:122]: Succes PUT should return 200

**Review by DelfiLT — CHANGES_REQUESTED** [review]

File: src/tests/super-admin.test.js [hunk 5]
```diff
@@ -1,3 +1,155 @@
...
+    expect(response.body.message).toBeDefined();
+    expect(response.body.error).toBeTruthy();
+  });
+
+  test('DELETE a non-existent ID status 404', async () => {
+    const response = await request(app).delete('/api/super-admin/122319de9d65e6b69280df4c9');
+    expect(response.status).toBe(400);
+    expect(response.body.message).toBeDefined();
+    expect(response.body.error).toBeTruthy();
+  });
```
[the remark this answers is not in the document]
Comment (ukiweskamp) [reply src/tests/super-admin.test.js:154]: Good point, thanks!

**Review by iaracriscenti — APPROVED** [review]
Nicely done! ⭐ ⭐ ⭐

**Review by DelfiLT — APPROVED** [review]

**Review by victoriassol — APPROVED** [review]

**Pull request merged**